independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/ExtensionCatalogResolver.java

```diff
@@ -327,7 +328,7 @@ public ExtensionCatalog resolveExtensionCatalog(String quarkusCoreVersion) throw
                 String commonQuarkusVersion = quarkusCoreVersion = release.getQuarkusCoreVersion();
                 int i = 1;
                 while (i < platformCatalog.getPlatforms().size() && commonQuarkusVersion != null) {
-                    final Platform p = platformCatalog.getPlatforms().get(i++);
+                    final Platform p = platformCatalog.get(i++);
```

Comment:
Same here. On every `get(index)` you are creating a an array just to get the item. Could you remove 'PlatformCatalog.get(index)` and iterate over `platformCatalog.getPlatforms()` instead?

Reply:
done